core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -3579,6 +3579,16 @@ private SqlDialect nonOrdinalDialect() {
     sql(retainLimitQuery).ok(retainLimitResult);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5504">[CALCITE-5504]
+   * Array literals are unparsed incorrectly for the spark dialect</a>.*/
```

Comment:
The java doc should stay the same as the Jira summary, Others are good to me.

Reply:
Do you mean the Description field?

```
They should be in the form: array(1, 2, 3)
However, they are unparsed as ARRAY[1, 2, 3] 

https://spark.apache.org/docs/latest/api/sql/#array
```

Reply:
change "Array literals are unparsed incorrectly for the spark dialect"  to "Array literals are unparsed incorrectly for the spark dialect" and rebase the commit info

Reply:
```
change 
"Array literals are unparsed incorrectly for the spark dialect" to 
"Array literals are unparsed incorrectly for the spark dialect" and rebase the commit info
```

Did I miss something? it's exactly the same.

Rebasing commits now

Reply:
ah you mean the commit title:

```
[CALCITE-5504] Unparse array value constructor for Apache Spark
```

vs jira

```
[CALCITE-5504] Array literals are unparsed incorrectly for the spark dialect
```

Reply:
here you go: https://github.com/apache/calcite/pull/3056/commits/8ba4534838688636d4c537e3867b99a04e297a88